typing/typedecl.ml

```diff
@@ -249,11 +251,27 @@ let transl_labels env closed lbls =
       lbls in
   lbls, lbls'
 
+let transl_types_gf env closed tyl =
+  let mk arg =
+    let cty = transl_simple_type env closed Global arg in
+    let gf = transl_global cty.ctyp_attributes in
+    (cty, gf)
+  in
+  let tyl_gfl = List.map mk tyl in
+  let tyl_gfl' =
+    List.map
+      (fun (cty,gf) ->
+         let ty = cty.ctyp_type in
+         (ty, gf)
+      )
+      tyl_gfl in
+  tyl_gfl, tyl_gfl'
```

Comment:
Style is a bit hard to read here. More usual would be:
```ocaml
  let tyl_gfl' =
    List.map
      (fun (cty, gf) ->
         let ty = cty.ctyp_type in
         ty, gf)
      tyl_gfl
  in
```
and you can probably shorten that to:
```ocaml
  let tyl_gfl' = List.map (fun (cty, gf) -> cty.ctyp_type, gf) tyl_gfl in
```